connect/hwinfo.go

```diff
@@ -0,0 +1,262 @@
+package connect
+
+import (
+	"bufio"
+	"bytes"
+	"net"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	cloudEx = `(?m)Version: .*(amazon)|Manufacturer: (Amazon)|Manufacturer: (Google)|Manufacturer: (Microsoft) Corporation`
+	cloudRe = regexp.MustCompile(cloudEx)
+)
+
+const (
+	archX86  = "x86_64"
+	archARM  = "aarch64"
+	archS390 = "s390x"
+)
+
+type hwinfo struct {
+	Hostname      string `json:"hostname"`
+	Cpus          int    `json:"cpus"`
+	Sockets       int    `json:"sockets"`
+	Hypervisor    string `json:"hypervisor"`
+	Arch          string `json:"arch"`
+	UUID          string `json:"uuid"`
+	CloudProvider string `json:"cloud_provider"`
+}
+
+func getHwinfo() (hwinfo, error) {
+	hw := hwinfo{}
+	var err error
+	hw.Arch, err = arch()
```

Comment:
with `err` defined above, these could be folded into the `if`s below.

Reply:
i.e. 
```
if hw.Arch, err = arch(); err != nil {
    return hwinfo{}, err
}

Reply:
done